e2e-tests/chain-simulator-e2e/event.processor.e2e.spec.ts

```diff
@@ -0,0 +1,195 @@
+
+      const pongTxHash = await sendTransaction({
+        chainSimulatorUrl: CHAIN_SIMULATOR_URL,
+        sender: ALICE_ADDRESS,
+        receiver: scAddress,
+        dataField: 'pong',
+        value: '0',
+        gasLimit: 20_000_000,
+      });
+      logMessage(`Called 'pong' function of the contract. Tx hash: ${pongTxHash}`);
+    }
+
+    await new Promise(resolve => setTimeout(resolve, 2000));
+
+    let numOfEventsReceived = 0;
+    let lastProcessedTimestamp = 0;
+    let counter = 0;
+    while (true) {
+      await eventProcessor.start(new EventProcessorOptions({
+        emitterAddresses: [scAddress],
+        eventIdentifiers: ['ping', 'pong'],
+        getLastProcessedTimestamp: async () => lastProcessedTimestamp,
+        elasticUrl: ELASTIC_SEARCH_URL,
+        onEventsReceived: async (highestTimestamp, events) => {
+          logMessage(`event processor received ${events.length} events with the highest timestamp ${highestTimestamp}`);
+          numOfEventsReceived += events.length;
+        },
+        setLastProcessedTimestamp: async (timestamp) => {
+          lastProcessedTimestamp = timestamp;
+        },
+        pageSize: 7,
+        delayBetweenRequestsInMilliseconds: 100,
+      }));
+
+      await new Promise(resolve => setTimeout(resolve, 100));
+      logMessage(`Running event processor #${counter + 1}`);
+
+      counter++;
+      if (counter > 10 || numOfEventsReceived === 2 * numPingPongs) {
+        break;
+      }
+    }
+
+    expect(numOfEventsReceived).toBe(numPingPongs * 2);
+  }, 100000);
+
+  it('should issue a token and event processor should receive esdt transfers', async () => {
+    await fundAddress(CHAIN_SIMULATOR_URL, BOB_ADDRESS);
+    const esdtIdentifier = await issueEsdt(new IssueEsdtArgs({
+      chainSimulatorUrl: CHAIN_SIMULATOR_URL,
+      issuer: BOB_ADDRESS,
+      tokenName: 'BobToken',
+      tokenTicker: 'BOB',
+    }));
+
+    const numTransfers = 20;
+    for (let i = 0; i < numTransfers; i++) {
+      const bobTransferTxHash = await transferEsdt(new TransferEsdtArgs({
+        chainSimulatorUrl: CHAIN_SIMULATOR_URL,
+        sender: BOB_ADDRESS,
+        receiver: ALICE_ADDRESS,
+        tokenIdentifier: esdtIdentifier,
+        plainAmountOfTokens: 5,
+      }));
+      logMessage(`Transferred 5 tokens to Bob. Tx hash: ${bobTransferTxHash}}`);
+      const carolTransferTxHash = await transferEsdt(new TransferEsdtArgs({
+        chainSimulatorUrl: CHAIN_SIMULATOR_URL,
+        sender: BOB_ADDRESS,
+        receiver: CAROL_ADDRESS,
+        tokenIdentifier: esdtIdentifier,
+        plainAmountOfTokens: 5,
+      }));
+      logMessage(`Transferred 5 tokens to Carol. Tx hash: ${carolTransferTxHash}}`);
+    }
+
+    await new Promise(resolve => setTimeout(resolve, 2000));
+
+    let numOfEventsReceived = 0;
+    let lastProcessedTimestamp = 0;
+    let counter = 0;
+    while (true) {
+      await eventProcessor.start(new EventProcessorOptions({
```

Comment:
* you have `eventProcessor.start` witch is used multiple time, maybe you can create a function to abstract the logic and that just call the function inside the testcase

Reply:
well, i call it with different arguments each time. I don't see a benefit extracting a function that just calls a function